### PR TITLE
Feature/dpll-pure-literals-elimination

### PR DESCRIPTION
**Pull request by w-disaster**
Pure literal elimination implementation.

### CODE REVIEW

File: src/main/scala/satify/update/dpll/DPLL.scala [hunk 1]
```diff
@@ -1,14 +1,19 @@
 package satify.update.dpll
 
-import satify.model.DecisionTree.*
-import satify.model.{CNF, Constraint, Decision, DecisionTree, PartialModel, Variable}
+import dotty.tools.dotc.transform.sjs.JSSymUtils.JSName.Literal
```
Comment (Mala1180) [inline src/main/scala/satify/update/dpll/DPLL.scala:3]: Wrong import?

File: src/main/scala/satify/update/dpll/DPLL.scala [hunk 2]
```diff
@@ -47,39 +52,48 @@ object DPLL:
             )
           case None => None
 
+    /** Branch on the pure literal branch first.
+      * @param dec the previous Decision.
+      * @return an Option containing a Branch if there's a pure literal, None otherwise.
+      */
+    def pureLitElDecision(dec: Decision): Option[Branch] =
```
Comment (Mala1180) [inline src/main/scala/satify/update/dpll/DPLL.scala:59]: You can do better on methods names

File: src/main/scala/satify/model/dpll/PartialModel.scala [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package satify.model.dpll
+
+import satify.model.Variable
+
+object OrderedSeq:
+  def seq[T](elems: T*)(using ordering: Ordering[T]): Seq[T] = elems.distinct.sorted(ordering)
+
+given Ordering[Variable] with
+  def compare(x: Variable, y: Variable): Int = (x, y) match
+    case (Variable(xName, _), Variable(yName, _)) => xName.compareTo(yName)
```
Comment (Mala1180) [inline src/main/scala/satify/model/dpll/PartialModel.scala:10]: I would put the given inside the object and then from outside I would use `import satify.model.dpll.OrderedSeq.given`
What do you think?

**Comment by w-disaster** [comment]
Thanks for the review!

**Review by paga16-hash — APPROVED** [review]

**Pull request merged**